pkg/controllers/dynakube/injection/reconciler.go

```diff
@@ -0,0 +1,141 @@
+package injection
+
+import (
+	"context"
+	goerrors "errors"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/status"
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta1/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/istio"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/version"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/namespace/ingestendpoint"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/namespace/initgeneration"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/namespace/mapper"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Reconciler interface {
+	Reconcile(ctx context.Context) error
+}
+
+type reconciler struct {
+	client            client.Client
+	apiReader         client.Reader
+	dynakube          *dynatracev1beta1.DynaKube
+	istioReconciler   istio.Reconciler
+	versionReconciler version.Reconciler
+}
+
+type ReconcilerBuilder func(client client.Client, apiReader client.Reader, dynakube *dynatracev1beta1.DynaKube, istioReconciler istio.Reconciler, versionReconciler version.Reconciler) Reconciler
+
+func NewReconciler(client client.Client, apiReader client.Reader, dynakube *dynatracev1beta1.DynaKube, istioReconciler istio.Reconciler, versionReconciler version.Reconciler) Reconciler {
+	return &reconciler{
+		client:            client,
+		apiReader:         apiReader,
+		dynakube:          dynakube,
+		istioReconciler:   istioReconciler,
+		versionReconciler: versionReconciler,
```

Comment:
On the one hand I would prefer that these 2 are not passed in, but created here
- so the `dyankube-controller` does not have to think about those
   -  the `istioReconciler` is a bit of an outlier, because the `dynakube-controller` will also use it, but maybe we can only reuse the `istioClient`?
   -  The current `istioReconciler` does comply with the `Reconcile` interface, because it was "quickly" converted into something that can support the reconcile of different sections, but that could possible be done better.

On the other hand, by not passing them in you would probably need to increase the params list

Reply:
Very good point, I adressed it in [this commit](https://github.com/Dynatrace/dynatrace-operator/pull/2736/commits/b721d3d23dd5429ee8dca1c827f673b3cef0fea0). It does increase the number of parameters (which made me even disable revive linter) but now the istio client is shared between active gate and injection, and in injection we create the reconcilers in the constructor.
Let me know your thoughts :+1: